jobspec/parse_task.go

```diff
@@ -494,8 +494,6 @@ func parseTemplates(result *[]*api.Template, list *ast.ObjectList) error {
 			ChangeMode:    stringToPtr("restart"),
 			Splay:         timeToPtr(5 * time.Second),
 			Perms:         stringToPtr("0644"),
-			Uid:           pointer.Of(-1),
-			Gid:           pointer.Of(-1),
```

Comment:
This breaks `TestParse/basic.hcl` but also looks off to me. Normally I'd expect to see an invalid value like this `-1` someplace where we _don't_ take a pointer, so that we can handle zero values that are meaningful (`0` means root in this case, which would be a bad default value). Are the downstream consumers of this value actually prepared to accept a nil pointer here?

Reply:
Great question! I did some digging, and I think this is the only place it's used (indirectly in its `structs/` form instead of this `api/` form): https://github.com/hashicorp/nomad/blob/v1.6.3/client/allocrunner/taskrunner/template/template.go#L727-L733

We actually explicitly changed code upstream from this to use `nil` instead of `-1` as part of 1.7, so I think this just got missed due to the missing `jobspec2` tests: https://github.com/hashicorp/nomad/pull/14203/files